engine/engine.go

```diff
@@ -251,7 +250,7 @@ func (e *EngineAPI) GetPayloadV3(ctx context.Context, payloadID engine.PayloadID
 		NoTxPool:             e.currentPayloadAttributes.NoTxPool,
 	})
 	if err != nil {
-		log.Panicf("failed to commit block: %v", err) // TODO error handling. An error here is potentially a big problem.
+		return nil, engine.GenericServerError.With(fmt.Errorf("build block: %v", err))
```

Comment:
Removing `log` is good, but let's keep the panic here for now. We can't really recover from an error here. We have a few work streams in progress to fix it, but it will take a while (db consistency, etc.).
```suggestion
		return panic(fmt.Errorf("build block: %v", err))
```